---
title: "Report - Heart"
output: 
  html_document: 
    theme: journal
date: "2025-03-13"
editor_options: 
  chunk_output_type: console
---

# Introduction
RNA-seq analysis of Control vs Ischemic heart tissue (n = 10 vs 13)

# Data Manipulation

## Processing

```{r BIOC}
# BIOC manager packages

if (!require("BiocManager", quietly = TRUE)) # only need one install
install.packages("BiocManager")

library(BiocManager)
library(DESeq2)
library(clusterProfiler)
library(org.Hs.eg.db)

```

```{r preamble}

# packages

remotes::install_github("vankesteren/firatheme")

library(tidyverse) # data wrangling, visualisation etc.
library(amap) # hierarchical clustering
library(reshape2) # data wrangling
library(ggrepel) # better plot labelling
library(wesanderson) # colour palettes
library(firatheme) # theme
library(showtext) # improved text formatting
library(cowplot) # plot production
library(ggtext)
```

## Install dfs

```{r df} 

# clear workspace
#rm(list = ls())

# load dataframes
annotations <- read.table("annotations.csv",
                          header = T, sep = "\t", row.names = 1)

de <- read.table("DE_control_vs_ischemic.csv",
                 header = T, sep = "\t", row.names = 1)

em <- read.table("EM.csv",
                 header = T, sep = "\t", row.names = 1)

ss <- read.table("sample_sheet.csv", header = T, sep = "\t", row.names = 1)

```

```{r tidy}

# summary
glimpse(annotations)

# merge dataframes to master df
master <- merge(em, annotations, by.x = 0, by.y = 0)

master <- merge(master, de, by.x = 1, by.y = 0) %>% 
  select(-1)

rownames(master) = master[,"symbol"]

colnames(master) # col 1:23 for expression matrix
em <- master[,1:23]

# sort rows by decreasing p adjusted
master %>% 
  na.omit %>% 
  arrange(p.adj) %>% 
  row.names() -> sort_order

# rows with NA values
# master %>% is.na %>% sum
# master %>% filter(is.na(p.adj) & p < 0.05) %>% count # 122 rows with p <0.05 omitted due to NA in p.adj

# reorder rows by descending p.adj
master <- master[sort_order,]

#rm(sort_order)
```

## Data Transformation

```{r transform}

# create columns for -log10p, mean expression, and significance
master %>%
  mutate(row_mean = rowMeans(select(.,1:23), na.rm = TRUE),
         mlog10p = -log10(p.adj),
         direction = factor(
           ifelse(log2fold >= 1 & p.adj <= 0.05, "Up",
                  ifelse(log2fold <= -1 & p.adj <= 0.05, "Down", "n.s")))) -> master

# Z-scaling of expression matrix - no. of sd's from mean
em %>% 
  t %>% 
  scale %>%  
  t %>%
  as.data.frame %>% 
  na.omit -> em_scaled
  
```

```{r sig subset}
# subset for sig genes
master_sig <- master %>% 
  filter(p.adj <= 0.05 & abs(log2fold) >= 1.0)

# extract gene column from master significance table, as a vector
sig_genes <- row.names(master_sig)

# index em_symbols by rows which correspond with values contained in sig_genes vector
#em_sig <- em[sig_genes,]

# z-scaling expression matrix
em_scaled_sig <- em_scaled[row.names(master_sig),]

# write df's in r environemnt to csv tables in the specified path "file", with separation by tab

#write.table(em_scaled_sig,
            #file = "M:\\L5\\Bioinformatics\\Tutorial 3 -11\\em_scaled_sig.csv",
            #sep = "\t")
```

# Plotting

```{r theme settings}
# font
font_add_google("Montserrat", "Montserrat") # google Montserrat font

showtext_auto() # use showtext package automatically

# colour palette
my_palette <- wes_palette(name = "Zissou1") 

my_palette_c <- wes_palette("Zissou1", 100, type = "continuous")

#print(my_palette)

theme_set(theme_fira() + 
            theme(text = element_text(family = "Montserrat"),
                  plot.title = element_markdown(),
                  axis.title.x = element_markdown(size = 16, family = "Montserrat"),
                  axis.title.y = element_markdown(size = 16,
                                                  family = "Montserrat"),
                  #axis.text.x = element_markdown(size = 16),
                  axis.text = element_markdown(size = 16, 
                                               family = "Montserrat")))

```

## Figure 1

### Figure 1a - Volcano Plot

```{r volcano plot}

plot_1a <- master %>% 
  ggplot(aes(log2fold, mlog10p, colour = direction)) +
  geom_point(alpha = 0.6, size = 2) +
  geom_hline(yintercept = -log10(0.05), linetype = 2, linewidth = 1) +
  geom_vline(xintercept = c(-1, 1), linetype = 2, linewidth = 1) +
  xlim(-5, 5) +
geom_text_repel(data = slice_head(master, n = 10),
                  aes(label = symbol),
                  show.legend = F, colour = "black", size = 5,
                  box.padding = 0.35,
                  point.padding = 0.5) +
  scale_colour_manual(values = my_palette[c(1,3,5)],
                      labels = c("↓", "n.s.", "↑")) +
  xlab("**log<sub>2</sub>FC**") +
  ylab("**-log<sub>10</sub>*p***") +
  labs(colour = "Change") +
  theme(legend.position = "top",
        legend.title = element_blank(),
        legend.text = element_text(size = 20, face = "bold")) +
        #legend.spacing.x = unit(6, "cm"),
        #plot.margin = margin(t = 40, r = 10, b = 10, l = 10)) +
  guides(colour = guide_legend(override.aes = list(size=5), 
                               nrow = 1, byrow = TRUE))

plot_1a

```

### Figure 1b - MA plot

```{r MA plot}
plot_1b <- master %>% 
  ggplot(aes(log10(row_mean), log2fold, colour = direction)) +
  geom_point(alpha = 0.6, size = 2) +
  geom_hline(yintercept = c(-1, 1), linetype = 2, linewidth = 1) +
  geom_text_repel(data = slice_head(master, n = 10),
                  aes(label = symbol),
                  show.legend = F, colour = "black", size = 5,
                  box.padding = 0.35,
                  point.padding = 0.5) +
  scale_colour_manual(values = my_palette[c(1,3,5)],
                      labels = c("↓", "n.s.", "↑")) +
  xlab("**log<sub>10</sub>Expression**") +
  ylab("**log<sub>2</sub>FC**") +
  theme(legend.position = "top",
        legend.title = element_blank(),
        legend.text = element_text(size = 20, face = "bold")) +
  guides(colour = guide_legend(override.aes = list(size=5), 
                               nrow = 1, byrow = TRUE))

plot_1b

```

### Figure 1c - Barplot

```{r barchart}

master %>% 
  group_by(direction) %>% 
  summarise(sum = n()) # 252 down, 182 up

plot_1c <- master %>% 
  filter(direction != "n.s") %>% 
  group_by(direction) %>% 
  summarise(number = n()) %>% 
  ggplot(aes(direction, number, fill = direction)) +
  geom_bar(stat = "identity") +
  geom_text(aes(label = number), vjust = 1.5, colour= "black", size = 10, family = "Montserrat") +
  scale_fill_manual(values = my_palette[c(1,5)]) +
  theme(legend.position = "none") +
  xlab("**Change**") +
  ylab("**Count**")

plot_1c

```

### Figure 1d - heatmap

```{r clusrtering in pipe chain}

hm_matrix <- as.matrix(em_scaled_sig[1:100,])

# hm_matrix_melted <- hm_matrix[order.dendrogram( # Order rows by dendrogram
#   hm_matrix %>%
#     Dist(method = "spearman") %>%
#     hclust(method = "average") %>%
#     as.dendrogram() %>%
#     reorder(0, FUN = "average")
# ), ] %>%
#   melt()

# rug to denote groups
rug <- data.frame(
  Sample = colnames(hm_matrix),   # Sample names
  Group = ss$sample_group           # Group labels (ICM vs CNT)
)

plot_1d <- hm_matrix[order.dendrogram( # order by dendrogram 
hm_matrix %>% 
  Dist(method = "euclidean") %>% # distance calc with euclidean
  hclust(method = "complete") %>% # hierarchical cluster by complete
  as.dendrogram %>% 
  reorder(0, FUN = "complete")
),] %>% 
  melt() %>% 
  ggplot(aes(Var2, Var1, fill = value)) +
  geom_tile(colour = NA) +
  scale_fill_gradientn(name = "Z-score", colours = my_palette_c) +
  ggnewscale::new_scale_fill() + # new fill for rug
  geom_tile(data = rug,   
            aes(Sample, y = -2, 
                fill = Group, 
                colour = Group), 
            inherit.aes = FALSE,
            height = 15,
            colour = "black",
            show.legend = F) +
  scale_fill_manual(values = c("ischemic" = my_palette[5], "control" = my_palette[4])) + # diff colours for cnt vs, icm
  theme(axis.text.y = element_blank(), 
        axis.ticks = element_blank(),
        legend.spacing.x = unit(0.01, "cm"),
        legend.position = "top",
        axis.title.x = element_blank(),
        axis.text.x = element_blank(),
        axis.title.y = element_blank(),
        legend.margin = margin(0, -5, 0, 0),
        legend.title = element_text(size = 14,
                                      face = "bold",
                                      vjust = 0.9),
          plot.margin = margin(t = 10, r = 10, b = 10, l = 10))

plot_1d

```

### Figure assembly


```{r figure 1}

windows()

fig1 <- plot_grid(plot_1a, plot_1b, plot_1c, plot_1d, 
                  labels = c("A","B", "C", "D"),
                  label_size = 16,
                  label_fontfamily = "Montserrat")

fig1

ggsave("figure1_bioinfo2.pdf", width = 12, height = 10, dpi = 600)

```

```{r clear workspace}
# clear figures from workspace
rm(list = c("plot_1a", "plot_1b", "plot_1c", "plot_1d", "fig1", "rug", "hm_matrix", "hm_matrix_melted"))

```

## Figure 2 - Quality Control

### 2a - Expression Density

```{r expression density final plot}

plot_2a <- em %>% 
  melt %>% 
  mutate(log10v = log10(value + 0.01),
    median = median(value),
              max = max(value),
             Type = ifelse(grepl("CNT", variable), "Control", "Ischaemic")) %>% 
  ggplot(aes(log10v, fill = Type)) +
    geom_density(alpha = 0.7, show.legend = T) +
    #geom_vline(aes(xintercept = mean(log10v)), linetype = 2, linewidth = 1, colour = my_palette[3]) +
  facet_wrap(~variable) +
  #facet_grid(rows = vars(variable), cols = vars(Type)) +
    #xlim(c(-5, 5)) +
    scale_fill_manual(values = my_palette[c(1,5)]) +
    xlab("**log<sub>10</sub>(Expression)**") +
    ylab("**Density**") +
  scale_x_continuous(limits = c(-5, 5), breaks = c(-5, 0, 5)) +
    theme(strip.text = element_blank(),
          axis.title.y = element_blank(),
          axis.title.x = element_markdown(hjust = 0.5),
          panel.border = element_rect(fill = NA),
          axis.text.y = element_text(size = 12),
          axis.text.x = element_text(size = 14),
          plot.margin = margin(t = 10, r = 10, b = 10, l = 10),
          legend.position = "top",
          legend.title = element_blank(),
          legend.text = element_text(size = 12, face = "bold"))

plot_2a

```

### 2b - correaltion scatterplot 

```{r correlation scatterplot}

mean_data <- master %>%
  rowwise() %>%  # Operate row by row
  mutate(
    cnt_mean = mean(c_across(starts_with("CNT")), na.rm = TRUE),
    icm_mean = mean(c_across(starts_with("ICM")), na.rm = TRUE)) %>% 
  ungroup()

pearson_corr <- round(cor(mean_data$cnt_mean, mean_data$icm_mean, method = "pearson"), 2)

cor.test(mean_data$cnt_mean, mean_data$icm_mean, method = "pearson") # r = 0.99,  p-value < 2.2e-16

pearson_corr

plot_2b <- mean_data %>% 
  ggplot(aes(log2(cnt_mean), log2(icm_mean))) +
  geom_point(colour = my_palette[1], 
             alpha = 0.25, 
             size = 2) +
  geom_smooth(se = F, 
              method = "lm", 
              colour = my_palette[5], 
              linewidth = 2,
              linetype = 2) +
  annotate(
    "richtext",
    x = mean(range(log2(mean_data$cnt_mean))),  # Center of x-axis
    y = max(log2(mean_data$icm_mean)) * 1.05,  # above y-axis max
    label = paste0("***R*<sup>2</sup>**: ", pearson_corr),
    size = 6
  ) +
  xlab("**log<sub>2</sub>(Control)**") +
  ylab("**log<sub>2</sub>(Ischaemic)**") +
  theme(plot.margin = margin(t = 10, r = 10, b = 10, l = 10),
        axis.title.y = element_markdown(hjust = 0.5),
        axis.title.x = element_markdown(hjust = 0.5))

plot_2b
```

### 2c - Heatmap - sample x sample

```{r heatmap}

plot_2c <- em_scaled_sig %>% 
    as.matrix() %>%
    cor(method = "spearman") %>%
    as.data.frame() %>% 
    rownames_to_column() %>%
    pivot_longer(cols = 2:24, names_to = "type", values_to = "Correlation") %>%
    ggplot(aes(rowname, type, fill = Correlation)) +
    geom_tile() +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
    scale_fill_gradientn(colours = my_palette_c) +
    theme(axis.title.x = element_blank(),
          axis.title.y = element_blank(), 
          axis.text.x = element_blank(),
          axis.text.y = element_blank(),
          legend.text = element_text(size = 12),
          legend.title = element_text(size = 14,
                                      face = "bold",
                                      vjust = 0.9, 
                                      hjust = 1),
          legend.position = "top",
          plot.margin = margin(t = 10, r = 10, b = 10, l = 10)) +
  guides(fill = guide_colourbar(barwidth = 10, 
                                 barheight = 1))

plot_2c

```

### 2d - PCA biplot

```{r PCA}

em_matrix <- as.matrix(sapply(em, as.numeric))

em.PCA <- prcomp(t(em_matrix), scale. = TRUE)

# is.numeric(em.PCA$x)

# summary(em.PCA)

# str(em.PCA$x)

# error occurs on occasion, clear workspace and rerun code
vars = apply(em.PCA$x, 2, var) # code to calculate percentage variance
```

```{r scree plot}

# scree plot - proportion of variance explained by PC
plot_2d <- vars %>% 
  data.frame %>%
  rownames_to_column(var = "PC") %>%
  rename(variance = 2) %>% 
  mutate(total = round(sum(variance), 4),
         prop = variance/total*100,
         #PC = factor(PC, levels = paste0("PC", 1:23),
         PC = factor(gsub("PC", "", PC), levels = 1:23)) %>%
  ggplot(aes(PC, prop)) +
  geom_hline(yintercept = 1, linetype = 2, linewidth = 1) +
  geom_line(group = 1, 
            colour = my_palette[1],
            linetype = 1,
            linewidth = 1) +
  geom_point(size = 4, 
             colour = my_palette[5]) +
  labs(x = "**Component**",
       y = "**Variance  (%)**") +
  theme(axis.text.x = element_text(size = 16,
                                   angle = 60,
                                   hjust = 1),
        axis.text.y = element_text(size = 16),
        axis.title.x = element_markdown(hjust = 0.5, 
                                        vjust = 1),
        axis.title.y = element_markdown(vjust = 0.5),
        plot.margin = margin(t = 10, r = 10, b = 10, l = 10))

plot_2d

```

### Figure assembly

```{r figure 2 assembly}

plot_grid(plot_2a, plot_2c, plot_2b, plot_2e, 
                  labels = c("A","B", "C", "D"),
                  label_size = 16,
                  label_fontfamily = "Montserrat")

ggsave("figure2_bioinfo2.pdf", width = 12, height = 10, dpi = 600)

# clear figures from workspace
#rm(list = c("plot_2a", "plot_2b", "plot_2c", "plot_2d"))

```

## Supplementary Figure 1

```{pca biplot theme}

theme_biplot <- theme(
        axis.text.x = element_text(size = 13),
        axis.text.y = element_text(size = 14),
        legend.position = "bottom",
        axis.title = element_text(face = "bold", hjust = 0.5),
        legend.title = element_blank(),
        legend.text = element_text(size = 14, 
                                   face = "bold"))

#rm(theme_biplot)
```

```{r labels}

# axis label - PC proportion of variance explained
prop_pc1 = round(vars["PC1"] / sum(vars),4) * 100
prop_pc2 = round(vars["PC2"] / sum(vars),4) * 100
x_axis_pc1 = paste("PC1" ,"(",prop_pc1,"%)",sep="")
y_axis_pc2 = paste("PC2" ,"(",prop_pc2,"%)",sep="")

prop_pc3 = round(vars["PC3"] / sum(vars),4) * 100
prop_pc4 = round(vars["PC4"] / sum(vars),4) * 100
x_axis_pc3 = paste("PC3" ,"(",prop_pc3,"%)",sep="")
y_axis_pc4 = paste("PC4" ,"(",prop_pc4,"%)",sep="")

prop_pc5 = round(vars["PC5"] / sum(vars),4) * 100
prop_pc6 = round(vars["PC6"] / sum(vars),4) * 100
x_axis_pc5 = paste("PC5" ,"(",prop_pc5,"%)",sep="")
y_axis_pc6 = paste("PC6" ,"(",prop_pc6,"%)",sep="")

prop_pc7 = round(vars["PC7"] / sum(vars),4) * 100
prop_pc8 = round(vars["PC8"] / sum(vars),4) * 100
x_axis_pc7 = paste("PC7" ,"(",prop_pc7,"%)",sep="")
y_axis_pc8 = paste("PC8" ,"(",prop_pc8,"%)",sep="")

```

```{r PCA biplot}
# PC1 vs PC2
em.PCA$x %>% as.data.frame() %>% 
  ggplot(aes(PC1, PC2)) +
  geom_point(size = 3, aes(colour = ss$sample_group)) +
  scale_colour_manual(labels = c("Control", "Ischaemic"),
                      values = my_palette[c(1,5)]) +
  xlab(x_axis_pc1) +
  ylab(y_axis_pc2) +
  labs(colour = "Sample") +
  theme_biplot -> S1a

# PC3 vs PC4
em.PCA$x %>% as.data.frame() %>% 
  ggplot(aes(PC3, PC4)) +
  geom_point(size = 3, aes(colour = ss$sample_group)) +
  scale_colour_manual(labels = c("Control", "Ischaemic"),
                      values = my_palette[c(1,5)]) +
  xlab(x_axis_pc3) +
  ylab(y_axis_pc4) +
  labs(colour = "Sample") +
  theme_biplot-> S1b

# PC5 vs PC6
em.PCA$x %>% as.data.frame() %>% 
  ggplot(aes(PC5, PC6)) +
  geom_point(size = 3, aes(colour = ss$sample_group)) +
  scale_colour_manual(labels = c("Control", "Ischaemic"),
                      values = my_palette[c(1,5)]) +
  xlab(x_axis_pc5) +
  ylab(y_axis_pc6) +
  labs(colour = "Sample") +
  theme_biplot -> S1c

# PC7 vs PC8
em.PCA$x %>% as.data.frame() %>% 
  ggplot(aes(PC7, PC8)) +
  geom_point(size = 3, aes(colour = ss$sample_group)) +
  scale_colour_manual(labels = c("Control", "Ischaemic"),
                      values = my_palette[c(1,5)]) +
  xlab(x_axis_pc7) +
  ylab(y_axis_pc8) +
  labs(colour = "Sample") +
  theme_biplot -> S1d

```

### Sup Fig 1

```{Sup figure 1 assembly}

plot_grid(S1a, S1b, S1c, S1d, 
                  labels = c("A","B", "C", "D"),
                  label_size = 16,
                  label_fontfamily = "Montserrat")

ggsave("supfig1_bioinfo2.pdf", width = 12, height = 10, dpi = 600)

```

## Figure 3 - Key genes

```{r heatmap}

em_scaled_sig %>% 
    as.matrix() %>%
    cor(method = "spearman") %>%
    as.data.frame() %>% 
    rownames_to_column() %>%
    pivot_longer(cols = 2:24, names_to = "type", values_to = "Correlation") %>%
    ggplot(aes(rowname, type, fill = Correlation)) +
    geom_tile() +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
    scale_fill_gradientn(colours = my_palette_c) +
    theme(axis.title.x = element_blank(),
          axis.title.y = element_blank(), 
          axis.text.x = element_blank(),
          axis.text.y = element_blank(),
          legend.text = element_text(size = 12),
          legend.title = element_text(size = 14,
                                      face = "bold",
                                      vjust = 0.9, 
                                      hjust = 1),
          legend.position = "top",
          plot.margin = margin(t = 10, r = 10, b = 10, l = 10)) +
  guides(fill = guide_colourbar(barwidth = 10, 
                                 barheight = 1))

# expression pattern for top 10 genes across 23 replicates - em unscaled
em_scaled[rownames(master[1:400,]),] %>%
    cor() %>%
    as.data.frame() %>%
    rownames_to_column() %>%
    pivot_longer(cols = -rowname, names_to = "V2", values_to = "corr") %>%
    ggplot(aes(V2, rowname, fill = corr)) +
    geom_tile() +
    scale_fill_gradientn(colours = my_palette_c) +
    theme_bw() +
    theme(axis.text.x = element_text(hjust = 1, angle = 60))

# correlation of expression pattern of top 10 genes across samples - em scaled
em_scaled[rownames(master[1:10,]),] %>%
  cor() %>% 
  as.data.frame() %>% 
  rownames_to_column() %>% 
  pivot_longer(cols = -rowname, names_to = "V2", values_to = "corr") %>% 
  ggplot(aes(V2, rowname, fill = corr)) +
  geom_tile() +
  scale_fill_gradientn(colours = my_palette_c) +
  labs(title = "Correlation of Expression Patterns across Samples",
       subtitle = "Top 10 most significant genes") +
  xlab("") +
  ylab("") +
  theme_bw() +
  theme(axis.text.x = element_text(hjust = 1, angle = 60))
  
```

## Figure 4 - Pathway analysis

## ORA - Biological Process

```{r ORA BP}

sig_genes_entrez = bitr(sig_genes, fromType = "ENSEMBL", toType = "ENTREZID", OrgDb = org.Hs.eg.db)
#rm(sig_genes_entrez)

# convert significant genes to entrez id
sig_genes_entrez = bitr(sig_genes, fromType = "SYMBOL", toType = "ENTREZID", OrgDb = org.Hs.eg.db)

# over-representation analysis
ora_results_bp = enrichGO(gene = sig_genes_entrez$ENTREZID, OrgDb = org.Hs.eg.db, readable = T, ont = "BP", pvalueCutoff = 0.05, qvalueCutoff = 0.10)

# plot
barplot(ora_results_bp, showCategory = 10, )

barplot()

ora_results_bp %>% View

dotplot(ora_results_bp, showCategory = 10)

goplot(ora_results_bp, showCategory = 3)

cnetplot(ora_results_bp, categorySize = "pvalue")

```

### Figure 3A - ORA BP - Barchart

```{r ggplot2}

ora_results_bp$pvalue

ORA_bp_barplot <- data.frame(p.adj = ora_results_bp$p.adjust,
           count = ora_results_bp$Count,
           set = ora_results_bp$Description) %>% 
  arrange(p.adj) %>%
  slice_head(n = 5) %>% 
  ggplot(aes(y = reorder(set, desc(count)), x = count, fill = p.adj)) +
  geom_bar(stat = "identity") +
  scale_fill_gradient(low = my_palette[1], high = my_palette[5], name = "*p*-value") +
  theme(axis.text.y = element_text(size = 12,
                                   hjust = 1,
                                   #face = "bold",
                                   margin = margin(r = 20)),
        axis.title.y = element_blank(),
        legend.title = element_markdown(face = "bold"),
        legend.text = element_text(face = "bold"),
        plot.margin = margin(t = 5, r = 10, b = 5, l = 20))

ORA_bp_barplot

```

### ORA BP

```{Pathway analysis part 2}

gene_sets_bp = ora_results_bp$geneID
description_bp = ora_results_bp$Description
p.adj_bp = ora_results_bp$p.adjust

```

### ORA BP Boxplot

```{r bp ORA}
ora_results_bp_table = data.frame(row.names = description_bp,
                               gene_set = gene_sets_bp,
                               p_adjusted = p.adj_bp)

ora_results_bp_table %>% View
enriched_gene_set_bp = as.character(ora_results_bp_table[1,"gene_set"])

ora_results_bp_table
candidate_genes_bp = unlist(strsplit(enriched_gene_set_bp, "/"))

candidates_bp <- master[candidate_genes_bp,]

ora_bp_box <- em_scaled[candidates_bp$symbol,] %>%
  rownames_to_column() %>%
  pivot_longer(cols = 2:24, names_to = "V1", values_to = "values") %>%
  mutate(V1 = factor(V1),
         rowname = factor(rowname)) %>%
  mutate(tissue = factor(ifelse(grepl("CNT", V1), "CNT", "ICM"))) %>% 
  ggplot(aes(tissue, values, fill = tissue)) +
  geom_violin() +
  geom_dotplot(binaxis= "y",
               stackdir = "center",
               dotsize = 1,
               fill = 1) +
  facet_wrap(~rowname) + 
  scale_fill_manual(values = my_palette[c(1,5)]) +
  theme(axis.text.x = element_blank(),
        strip.text = element_text(face = "bold"))

enriched_gene_set_bp
ora_bp_box

ggsave()

View(enriched_gene_set)

ridgeplot(candidate_genes)

ora_results_mf_table
```

```{r ORA MF}

gene_sets_mf = ora_results_mf$geneID
description_mf = ora_results_mf$Description
p.adj_mf = ora_results_mf$p.adjust

ora_mf_gene_names <- ora_results_mf$FoldEnrichment
```

```{r ORA MF}

ora_results_mf_table = data.frame(row.names = description_mf,
                               gene_set = gene_sets_mf,
                               p_adjusted = p.adj_mf)

ora_results_mf_table %>% View()

#ora_results_mf_table %>% View

enriched_gene_set_mf = as.character(ora_results_mf_table[1,"gene_set"])

candidate_genes_mf = unlist(strsplit(enriched_gene_set_mf, "/"))

candidates_mf <- master[candidate_genes_mf,]
```

## Figure 4 - GO violins, boxplot top 10 etc.

```{R boxplot MF ORA}
plot_4a <- em_scaled[candidates_mf$symbol,] %>%
  rownames_to_column() %>%
  pivot_longer(cols = 2:24, names_to = "V1", values_to = "values") %>%
  mutate(V1 = factor(V1),
         rowname = factor(rowname)) %>%
  mutate(tissue = factor(ifelse(grepl("CNT", V1), "CNT", "ICM"))) %>% 
  ggplot(aes(tissue, values, fill = tissue)) +
  geom_violin() +
  geom_dotplot(binaxis= "y",
               stackdir = "center",
               dotsize = 1,
               fill = 1) +
  facet_wrap(~rowname, nrow = 2) + 
  labs(y = "Z-score") +
  scale_fill_manual(values = my_palette[c(1,5)], labels = c("Control", "Ischaemic")) +
  theme(axis.text.x = element_blank(),
        strip.text = element_text(face = "bold", size = 14),
        axis.title.x = element_blank(),
        legend.position = "right",
        legend.title = element_blank(), 
        legend.text = element_text(size = 14, face = "bold"),
        panel.border = element_rect(fill = NA))

plot_4a

```

### top 10 boxplot of z-score expression ####

```{r}
plot_4b <- em_scaled[rownames(master_sig[1:10,]),] %>%
  t() %>% 
  as.data.frame() %>%
  rownames_to_column(var = "samp") %>%
  pivot_longer(!samp, 
               names_to = "gene", 
               values_to = "value") %>% 
  mutate(status = factor(ifelse(grepl("CNT", samp), "Control", "Ischemic"))) %>%  
  ggplot(aes(status, value, fill = status)) +
  geom_boxplot(alpha = 0.5, linewidth = 0.5) +
  geom_jitter(alpha = 1, width = 0.2, size = 2, show.legend = F) +
  ylab("**Z-score**") +
  scale_fill_manual(values = my_palette[c(1,5)], labels = c("Control", "Ischaemic")) +
  facet_wrap(~gene, nrow = 5) + 
  theme(strip.text = element_text(face = "bold", size = 16),
          axis.title.x = element_blank(),
          panel.border = element_rect(fill = NA),
          axis.text.y = element_text(),
          axis.text.x = element_blank(),
          plot.margin = margin(t = 10, r = 10, b = 10, l = 10),
          legend.position = "bottom",
          legend.title = element_blank(),
          legend.text = element_text(size = 12, face = "bold"))
plot_4b

```

```{r CNETplot}

```{r ORA MF}
ora_results_mf = enrichGO(gene = sig_genes_entrez$ENTREZID, OrgDb = org.Hs.eg.db, readable = T, ont = "MF", pvalueCutoff = 0.05, qvalueCutoff = 0.10)

ora_results_mf %>% View
# plot
barplot(ora_results_mf, showCategory = 10)

dotplot(ora_results_mf, showCategory = 10)

goplot(ora_results_mf, showCategory = 10)

gcnet = cnetplot(ora_results_mf, categorySize = "pvalue")

gcnet

```

```{r}
gcnet = cnetplot(ora_results_mf, 
                 categorySize = "pvalue", 
                 size_item = 1, 
                 size_category = 2, 
                 layout = "kk",
                 cex_label_category = 2,
                 cex_label_gene = 2, showCategory = 5)      

gcnet

ora_results_mf_table
goplot_mf <- goplot(ora_results_mf, showCategory = 10)
```

### Figure assembly
```{Figure assembly}

dev.new()

plot_grid(plot_4a, gcnet,
                  labels = c("A","B"),
                  nrow = 2,
                  label_size = 16,
                  label_fontfamily = "Montserrat")
dev.new()
plot_grid(gcnet, goplot_mf,
                  labels = c("A","B"),
                  nrow = 2,
                  label_size = 16,
                  label_fontfamily = "Montserrat", )

gcnet

ggsave("figure_4_bioinfo10.pdf", width = 12, height = 10, dpi = 600)

```

# Additional Figures

### BOXPLOT MF ORA

```{r ggplot2}

ora_results_mf$pvalue

ORA_mf_barplot <- data.frame(p.adj = ora_results_mf$p.adjust,
           count = ora_results_mf$Count,
           set = ora_results_mf$Description) %>% 
  arrange(p.adj) %>%
  slice_head(n = 5) %>% 
  ggplot(aes(y = reorder(factor(set), desc(count)), x = count, fill = p.adj)) +
  geom_bar(stat = "identity") +
  scale_fill_gradient(low = my_palette[1], high = my_palette[5], name = "*p*-value") +
  theme(axis.text.y = element_text(size = 12,
                                   hjust = 1,
                                   #face = "bold",
                                   margin = margin(r = 20)),
        axis.title.y = element_blank(),
        legend.title = element_markdown(face = "bold"),
        legend.text = element_text(face = "bold"),
        plot.margin = margin(t = 5, r = 10, b = 5, l = 20))

ORA_mf_barplot

```

### Misc Figure

```{r}

plot_grid(ORA_mf_barplot, ORA_bp_barplot, 
                  labels = c("A","B"),
                  nrow = 2,
                  label_size = 16,
                  label_fontfamily = "Montserrat")

plot_grid(ORA_mf_barplot, plot_4a, 
                  labels = c("A","B"),
                  nrow = 2,
                  label_size = 16,
                  label_fontfamily = "Montserrat")

ggsave("figure_3_bioinfo10.pdf", width = 12, height = 10, dpi = 600)
```

```{r enriched gene set 2}

enriched_gene_set_2 = as.character(ora_results_mf_table[2,"gene_set"])

candidate_genes_2 = unlist(strsplit(enriched_gene_set_2, "/"))

top_10_candidates_2 <- master[candidate_genes_2,]


em_scaled[top_10_candidates_2$symbol,] %>%
  rownames_to_column() %>%
  pivot_longer(cols = 2:24, names_to = "V1", values_to = "values") %>%
  mutate(V1 = factor(V1),
         rowname = factor(rowname)) %>%
  mutate(tissue = factor(ifelse(grepl("CNT", V1), "CNT", "ICM"))) %>% 
  ggplot(aes(tissue, values, fill = tissue)) +
  geom_boxplot() +
  facet_wrap(~rowname) + 
  scale_fill_manual(values = my_palette[c(1,5)]) +
  theme(axis.text.x = element_blank(),
        strip.text = element_text(face = "bold"))

ora_results_table %>% View
```

```{r enriched gene set 3}

enriched_gene_set_3 = as.character(ora_results_mf_table[3,"gene_set"])

candidate_genes_3 = unlist(strsplit(enriched_gene_set_3, "/"))

top_10_candidates_3 <- master[candidate_genes_3,]


em_scaled[top_10_candidates_3$symbol,] %>%
  rownames_to_column() %>%
  pivot_longer(cols = 2:24, names_to = "V1", values_to = "values") %>%
  mutate(V1 = factor(V1),
         rowname = factor(rowname)) %>%
  mutate(tissue = factor(ifelse(grepl("CNT", V1), "CNT", "ICM"))) %>% 
  ggplot(aes(tissue, values, fill = tissue)) +
  geom_boxplot() +
  facet_wrap(~rowname) + 
  scale_fill_manual(values = my_palette[c(1,5)]) +
  theme(axis.text.x = element_blank(),
        strip.text = element_text(face = "bold"))

```

```{r gene set 4}

enriched_gene_set_4 = as.character(ora_results_table[4,"gene_set"])

candidate_genes_4 = unlist(strsplit(enriched_gene_set_4, "/"))

top_10_candidates_4 <- master[candidate_genes_4,]


em_scaled[top_10_candidates_4$symbol,] %>%
  rownames_to_column() %>%
  pivot_longer(cols = 2:24, names_to = "V1", values_to = "values") %>%
  mutate(V1 = factor(V1),
         rowname = factor(rowname)) %>%
  mutate(tissue = factor(ifelse(grepl("CNT", V1), "CNT", "ICM"))) %>% 
  ggplot(aes(tissue, values, fill = tissue)) +
  geom_boxplot() +
  facet_wrap(~rowname) + 
  scale_fill_manual(values = my_palette[c(1,5)]) +
  theme(axis.text.x = element_blank(),
        strip.text = element_text(face = "bold"))

```

```{r gene set 5}

enriched_gene_set_5 = as.character(ora_results_table[5,"gene_set"])

candidate_genes_5 = unlist(strsplit(enriched_gene_set_5, "/"))

top_10_candidates_5 <- master[candidate_genes_5,]


em_scaled[top_10_candidates_5$symbol,] %>%
  rownames_to_column() %>%
  pivot_longer(cols = 2:24, names_to = "V1", values_to = "values") %>%
  mutate(V1 = factor(V1),
         rowname = factor(rowname)) %>%
  mutate(tissue = factor(ifelse(grepl("CNT", V1), "CNT", "ICM"))) %>% 
  ggplot(aes(tissue, values, fill = tissue)) +
  geom_boxplot() +
  facet_wrap(~rowname) + 
  scale_fill_manual(values = my_palette[c(1,5)]) +
  theme(axis.text.x = element_blank(),
        strip.text = element_text(face = "bold"))

gse_results %>% View

```

```{r enriched ontology}

enriched_go = as.character(ora_results_table[1,"gene_set"])

```

### GSEA

```{r GSEA input}

gsea_input = master$log2fold

names(gsea_input) = row.names(master)

gsea_input = na.omit(gsea_input)

gsea_input = sort(gsea_input, decreasing = TRUE)

View(gsea_input)
```

### GSEA MF
```{r gsea}
gse_results_mf = gseGO(geneList = gsea_input,
                    ont = "MF",
                    keyType = "SYMBOL",
                    nPerm = 10000, 
                    minGSSize = 3, 
                    maxGSSize = 800, 
                    pvalueCutoff = 0.05, 
                    verbose = TRUE, 
                    OrgDb = org.Hs.eg.db, 
                    pAdjustMethod = "none")

ridgeplot(gse_results_mf, showCategory = 10)

enrichplot::dotplot(gse_results_bp)

gsea_results_bp_table %>% View

```

```{r GSEA MF}

gsea_results_mf_table = data.frame(row.names = gse_results_mf$Description,
                                   p.adj = gse_results_mf$p.adjust,
                                   NES = gse_results_mf$NES,
                                   gse_results_mf$core_enrichment)

gsea_results_mf_table %>% View

```

```{r}
enriched_gsea_mf = as.character(gsea_results_mf_table[1,"gse_results_mf.core_enrichment"])

gsea_results_mf_table[1:5, "gse_results_mf.core_enrichment"]

candidate_genes_gsea_bp = unlist(strsplit(enriched_gsea_bp, "/"))

candidates_gsea_bp <- master[candidate_genes_gsea_bp,]

em_scaled[candidates_gsea_bp$symbol,] %>%
  rownames_to_column() %>%
  pivot_longer(cols = 2:24, names_to = "V1", values_to = "values") %>%
  mutate(V1 = factor(V1),
         rowname = factor(rowname)) %>%
  mutate(tissue = factor(ifelse(grepl("CNT", V1), "CNT", "ICM"))) %>% 
  ggplot(aes(tissue, values, fill = tissue)) +
  geom_boxplot() +
  facet_wrap(~rowname) + 
  scale_fill_manual(values = my_palette[c(1,5)]) +
  theme(axis.text.x = element_blank(),
        strip.text = element_text(face = "bold"))
```


### GSEA BP

```{r gsea}
gse_results_bp = gseGO(geneList = gsea_input,
                    ont = "BP",
                    keyType = "SYMBOL",
                    nPerm = 10000, 
                    minGSSize = 3, 
                    maxGSSize = 800, 
                    pvalueCutoff = 0.05, 
                    verbose = TRUE, 
                    OrgDb = org.Hs.eg.db, 
                    pAdjustMethod = "none")

ridgeplot(gse_results, showCategory = 10)
```

```{r GSEA BP}

gsea_results_bp_table = data.frame(row.names = gse_results_bp$Description,
                               p.adj = gse_results_bp$p.adjust,
                               NES = gse_results_bp$NES,
                               gse_results_bp$core_enrichment)

gsea_results_bp_table %>% View

barplot(gse_results_bp, showCategory = 10)

dotplot(gse_results_bp, showCategory = 10)

goplot(gse_results_bp, showCategory = 3)

cnetplot(ora_results_bp, categorySize = "pvalue")

```

### GSEA BP

```{r}
enriched_gsea_bp = as.character(gsea_results_bp_table[1,"gse_results_bp.core_enrichment"])

gsea_results_bp_table[3, "gse_results_bp.core_enrichment"]

candidate_genes_gsea_bp = unlist(strsplit(enriched_gsea_bp, "/"))

candidates_gsea_bp <- master[candidate_genes_gsea_bp,]

em_scaled[candidates_gsea_bp$symbol,] %>%
  rownames_to_column() %>%
  pivot_longer(cols = 2:24, names_to = "V1", values_to = "values") %>%
  mutate(V1 = factor(V1),
         rowname = factor(rowname)) %>%
  mutate(tissue = factor(ifelse(grepl("CNT", V1), "CNT", "ICM"))) %>% 
  ggplot(aes(tissue, values, fill = tissue)) +
  geom_boxplot() +
  facet_wrap(~rowname) + 
  scale_fill_manual(values = my_palette[c(1,5)]) +
  theme(axis.text.x = element_blank(),
        strip.text = element_text(face = "bold"))


```

### GSEA MF

```{R}
gse_results_mf = gseGO(geneList = gsea_input,
                    ont = "BP",
                    keyType = "SYMBOL",
                    nPerm = 10000, 
                    minGSSize = 3, 
                    maxGSSize = 800, 
                    pvalueCutoff = 0.05, 
                    verbose = TRUE, 
                    OrgDb = org.Hs.eg.db, 
                    pAdjustMethod = "none")

ridgeplot(gse_results, showCategory = 10)
```

```{r GSEA BP}

gsea_results_bp_table = data.frame(row.names = gse_results_bp$Description,
                               p.adj = gse_results_bp$p.adjust,
                               NES = gse_results_bp$NES,
                               gse_results_bp$core_enrichment)

gsea_results_bp_table %>% View

```

## String

```{r}

BiocManager::install("STRINGdb")
library(STRINGdb)

cand_gene_table = data.frame(candidate_genes)

names(cand_gene_table) = "gene"

string_db = STRINGdb$new( version="11.5", species=9606, score_threshold=200, network_type="full", input_directory="")

string_mapped = string_db$map(cand_gene_table, "gene", removeUnmappedRows = TRUE ) 

string_db$plot_network(string_mapped)

```
